cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/JsonMenu/PluginJsonImporter/index.tsx

```diff
@@ -0,0 +1,128 @@
+import { green } from '@material-ui/core/colors';
+import { StyleRules } from '@material-ui/core/styles';
+import CheckIcon from '@material-ui/icons/Check';
+import SaveIcon from '@material-ui/icons/Save';
+import clsx from 'clsx';
+import { JSONStatusMessage } from 'components/PluginJSONCreator/Create/Content/JsonMenu';
+import { ICreateContext } from 'components/Replicator/Create';
+import React from 'react';
+
+const styles = (theme): StyleRules => {
+  return {
+    buttonSuccess: {
+      backgroundColor: green[500],
+      '&:hover': {
+        backgroundColor: green[700],
+      },
+    },
+    buttonProgress: {
+      color: green[500],
+      position: 'absolute',
+      top: '50%',
+      left: '50%',
+      marginTop: -12,
+      marginLeft: -12,
+    },
+    fileInput: {
+      display: 'none',
+    },
+  };
+};
+
+interface IPluginJSONImporterProps extends WithStyles<typeof styles>, ICreateContext {
+  populateImportResults: (filename: string, fileContent: string) => void;
+  JSONStatus: JSONStatusMessage;
+}
+
+const PluginJSONImporterView: React.FC<IPluginJSONImporterProps> = ({
+  classes,
+  populateImportResults,
+  JSONStatus,
+}) => {
+  const [loading, setLoading] = React.useState(false);
+  const [success, setSuccess] = React.useState(false);
+
+  const buttonClassname = clsx({
+    [classes.buttonSuccess]: success,
+  });
+
+  React.useEffect(() => {
+    if (JSONStatus === JSONStatusMessage.Success) {
+      setSuccess(true);
+    } else {
+      setSuccess(false);
+    }
+  }, [JSONStatus]);
+
+  function processFileUpload() {
+    return (e) => {
+      if (!loading) {
+        setLoading(true);
+      }
+      const files = e.target.files;
+      if (files.length > 0) {
+        const filename = files[0].name;
+        const filenameWithoutExtension =
+          filename.substring(0, filename.lastIndexOf('.')) || filename;
+        const reader = new FileReader();
+        reader.readAsText(files[0]);
+        let fileContent;
+        reader.onload = (r) => {
+          fileContent = r.target.result;
+          renderFileContent(filenameWithoutExtension, fileContent);
+        };
+      } else {
+        setLoading(false);
+      }
+    };
+  }
+
+  async function renderFileContent(filename, fileContent) {
+    await populateImportResults(filename, fileContent);
+    setLoading(false);
+  }
+
+  return (
+    <div>
+      <input
+        accept="json/*"
+        id="raised-button-file"
+        type="file"
+        className={classes.fileInput}
+        onChange={processFileUpload()}
+      />
+      <label htmlFor="raised-button-file">
+        <Button
+          aria-label="save"
+          component="span"
+          color="primary"
+          disabled={loading}
+          className={buttonClassname}
+        >
+          {success ? <CheckIcon /> : <SaveIcon />}
+        </Button>
+        {loading && <CircularProgress size={24} className={classes.buttonProgress} />}
```

Comment:
nit: Use `LoadingSVG` icon

Reply:
I ended up removing the `CircularProgress` since we're now remounting component on file import, as per your suggestion from https://github.com/cdapio/cdap/pull/12229#discussion_r434321506